contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -328,31 +428,255 @@ abstract contract BaseHarvester is Governable {
     }
 
     /**
-     * @dev Swap all supported swap tokens for stablecoins via Uniswap. And send the incentive part
-     * of the rewards to _rewardTo address.
-     * @param _rewardTo Address where to send a share of harvest rewards to as an incentive
-     *      for executing this function
+     * @dev Swap a reward token for the base token on the configured
+     *      swap platform. The token must have a registered price feed
+     *      with the price provider
+     * @param _swapToken Address of the token to swap
+     * @param _rewardTo Address where to send the share of harvest rewards to
      */
-    function _swap(address _rewardTo) internal {
-        address[] memory allStrategies = IVault(vaultAddress)
-            .getAllStrategies();
-
-        for (uint256 i = 0; i < allStrategies.length; i++) {
-            IStrategy strategy = IStrategy(allStrategies[i]);
-            address[] memory rewardTokenAddresses = strategy
-                .getRewardTokenAddresses();
-
-            for (uint256 j = 0; j < rewardTokenAddresses.length; j++) {
-                _swap(rewardTokenAddresses[j], _rewardTo);
-            }
+    function _swap(address _swapToken, address _rewardTo) internal virtual {
+        RewardTokenConfig memory tokenConfig = rewardTokenConfigs[_swapToken];
+
+        /* This will trigger a return when reward token configuration has not yet been set
+         * or we have temporarily disabled swapping of specific reward token via setting
+         * doSwapRewardToken to false.
+         */
+        if (!tokenConfig.doSwapRewardToken) {
+            return;
```

Comment:
would be so nice to bubble this information up somehow. An error might be wrong, if multiple tokens are being harvested. Maybe an event saying token swap disabled for a specific token?

Reply:
I'm not entirely sure if we should do that. I don't think we would have any use of that event on analytics side either?

But it won't hurt to add it, can do it if you think there would be a need for it